process/process_posix.go

```diff
@@ -78,23 +79,35 @@ func PidExistsWithContext(ctx context.Context, pid int32) (bool, error) {
 	if err != nil {
 		return false, err
 	}
-	err = proc.Signal(syscall.Signal(0))
-	if err == nil {
-		return true, nil
-	}
-	if err.Error() == "os: process already finished" {
-		return false, nil
-	}
-	errno, ok := err.(syscall.Errno)
-	if !ok {
-		return false, err
-	}
-	switch errno {
-	case syscall.ESRCH:
-		return false, nil
-	case syscall.EPERM:
-		return true, nil
+
+	if ! strings.HasPrefix(common.HostProc(),"/proc") { //Means that proc is mounted not to /proc, and we highly
+		// likely running inside container with a different process namespace (by default), so we check pid existence
+		// based on /<HOST_PROC>/proc/<PID> folder
+		file, err := os.Open(common.HostProc(strconv.Itoa(int(pid))))
```

Comment:
Use `os.Stat` instead as in this commit https://github.com/shirou/gopsutil/commit/258343806a4d3002c09ae9632715b5364df840e0